lib/bundler/metrics.rb

```diff
@@ -0,0 +1,141 @@
+# frozen_string_literal: true
+
+require "time"
+require "securerandom"
+require "digest"
+
+module Bundler
+  class Metrics
+    attr_accessor :metrics_hash, :path
+
+    def initialize
+      @path = Pathname.new(File.join(Bundler.user_home.join(".bundle"), "metrics.yml"))
+      add_metrics
```

Comment:
Don't call `add_metrics` here when the object is created, instead create an empty storage for metrics to be placed into. Like this: `@metrics = Queue.new`

Reply:
Any specific reason for a queue rather than an array? it seems like my data isn't getting pushed into the queue.

Reply:
Unless I missed something (and maybe I did! 😅) each metric entry is a hash, like this: `{timestamp: "2019-18-05T12:12:12.12", command: "install"}` etc. That means we need an array to hold each entry. If we're using an array, I think we want to use `Queue` instead of `Array` because queues are thread-safe, and that will mean we can later on switch to recording and sending metrics in another thread. I think we will want to switch to reading/writing/sending metrics in a background thread eventually, to keep from slowing down the other stuff Bundler is doing.